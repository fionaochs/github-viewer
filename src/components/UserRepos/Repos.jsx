import React from 'react';
import PropTypes from 'prop-types';
import Repo from './Repo.jsx';

const Repos = ({ repos }) => {
  const repoElements = repos.map(repo => (
    <li key={repo.id}>
      <Repo {...repo} />
    </li>
  ));

  return (
    <ul>
      {repoElements}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    userName: PropTypes.string.isRequired,
    repoName: PropTypes.string.isRequired,
    pr: PropTypes.string.isRequired,
    issues: PropTypes.string.isRequired,
  })).isRequired
};

export default Repos;

