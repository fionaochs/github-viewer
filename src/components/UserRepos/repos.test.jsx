import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos', () => {
  it('match a repos snapshot', () => {
    const repos = [
      {
        id: 1,
        userName: 'me',
        repoName: 'repo',
        pr: 'pr',
        issues: 'issues'
      },
      {
        id: 2,
        userName: 'me',
        repoName: 'repo',
        pr: 'pr',
        issues: 'issues'
      }
    ];
    const wrapper = shallow(<Repos repos={repos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
