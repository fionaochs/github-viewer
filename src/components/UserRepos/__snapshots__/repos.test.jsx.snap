// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos match a repos snapshot 1`] = `
<ul>
  <li
    key="1"
  >
    <Repo
      id={1}
      issues="issues"
      pr="pr"
      repoName="repo"
      userName="me"
    />
  </li>
  <li
    key="2"
  >
    <Repo
      id={2}
      issues="issues"
      pr="pr"
      repoName="repo"
      userName="me"
    />
  </li>
</ul>
`;
